### PR TITLE
Issue 1362/total flagged bug

### PR DESCRIPTION
**Pull request by pbn4**
# Pull Request Template

## Issue

Addresses # (#1362)
- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This PR addresses two problems:

- incorrectly displaying totalFlagged (backend ignored listingId query param when counting totalFlagged in AFS table)
- a race condition when two applications were added in similar time, resulting in one of them not being captured by an AFS table (e.g. a duplicate set contained one application less than it should). The new implementation now runs the entire application insert and AFS creation as a transaction with serializable isolation level. This now leads to potential read/write dependencies between transactions and PSQL aborting one of the ongoing transactions (in a case where two users submit an application in the same time). I have implemented basic expontential backoff retry logic with `async-retry` package, but the maximum number of retries is 6, with 200ms backoff as a starting base and with expontent factor equal to 2. So it's a maximum wait of `200+400+800+1600+3200 + 6400 = 12600ms` otherwise backend will throw a 500 error and abort both application and AFS creation. 

@seanmalbert Maybe we should consider returning something like `429 too many requests` instead of a 500 for the frontend to display an appropriate message then such transaction dependency happens? 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [ ] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

Please describe the tests that you ran to verify your changes. Provide instructions so we can review. Please also list any relevant details for your test configuration
- [ ] Desktop View
- [ ] Mobile View
- [ ] Test A
- [ ] Test B

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have updated the changelog to include a description of my changes

### CODE REVIEW

**Review by dominikx96 — APPROVED** [review]
Tested it locally (manually) and I don't see any issues, focused just on the frontend part.

**Comment by pbn4** [comment]
@seanmalbert Can we merge or do you still plan to review?

File: backend/core/src/application-flagged-sets/application-flagged-sets.service.ts [hunk 1]
```diff
@@ -56,63 +66,87 @@ export class ApplicationFlaggedSetsService {
...
-    await this.afsRepository.save(afs)
-    return afs
+      for (const appNotToBeResolved of appsNotToBeResolved) {
+        appNotToBeResolved.markedAsDuplicate = false
+        await transApplicationsRepository.save(appNotToBeResolved)
+      }
+      appsToBeResolved.forEach((app) => (app.markedAsDuplicate = true))
+      await transAfsRepository.save(afs)
+      return afs
+    })
```
Comment (seanmalbert) [inline backend/core/src/application-flagged-sets/application-flagged-sets.service.ts:102]: I know this logic already existed, but these separate iterations could probably be condensed down into one. I'm not sure how large these record sets in this case get though, so not sure if it's worth any savings.

**Review by seanmalbert — APPROVED** [review]
@pbn4 ,
This looks good. A 429 may be better than a 500 in this case, but I don't know if that message should be getting back to the user. Ideally the applicant doesn't lose any data and they're told to try again in a few seconds unless the problem persists. Also, do you have a good sense on how frequent this issue has been?

**Comment by pbn4** [comment]
> Also, do you have a good sense on how frequent this issue has been?

In a test sending two requests in parallel it occured almost 100% of the time. 

I added 429 throwing and collapsed DB writes where you have pointed out.

**Pull request merged**